x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/TrainedModelDeploymentTask.java

```diff
@@ -77,10 +77,10 @@ public void updateNumberOfAllocations(int numberOfAllocations) {
         params = new TaskParams(
             params.getModelId(),
             params.getModelBytes(),
-            numberOfAllocations,
             params.getThreadsPerAllocation(),
+            numberOfAllocations,
```

Comment:
It seems that this mistake came from the fact that the public and the private constructor in `TaskParams` have a different order of parameters (`numberOfAllocations` vs `threadsPerAllocation`).
You may want to align these to avoid similar mistakes in the future.

Reply:
Good point. Done!